web/src/InstallationProgress.jsx

```diff
@@ -34,11 +34,9 @@ function InstallationProgress() {
   const [progress, setProgress] = useState({});
 
   useEffect(() => {
-    return client.onPropertyChanged((_path, _iface, signal, args) => {
-      const iface = "org.opensuse.DInstaller.Manager1";
-      const [input_iface, changed] = args;
-      if (input_iface === iface && "Progress" in changed) {
-        const [title, steps, step, substeps, substep] = changed.Progress.v.map(pr => pr.v);
+    return client.manager.onChange(changes => {
```

Comment:
maybe I would go one step further and have method `onProgress` that just get that progress as param?

Reply:
It might be possible, I am not against that. But we might end up with one `onXXX` method for each property. Alternatively, we could have an argument to filter by property name. But as it is rather straight-forward, we decided to keep it as it is until the API evolves a little bit (based on our use cases).

Reply:
Actually, we could do both cases: adding the "name" filtering and defining `onProgress` on top of that.